flexiblepricing/mail_api.py

```diff
@@ -0,0 +1,89 @@
+"""
+Utils for mail
+"""
+import logging
+
+from django.core.exceptions import ValidationError
+from mitol.mail.api import get_message_sender
+
+from courses.models import CourseRun
+from flexiblepricing.constants import (
+    FLEXIBLE_PRICE_APPROVAL_MESSAGE,
+    FLEXIBLE_PRICE_APPROVAL_SUBJECT,
+    FLEXIBLE_PRICE_DOCUMENTS_RECEIVED_MESSAGE,
+    FLEXIBLE_PRICE_DOCUMENTS_RESET_MESSAGE,
+    FLEXIBLE_PRICE_RESET_SUBJECT,
+    FLEXIBLE_PRICE_DOCUMENTS_RECEIVED_SUBJECT,
+    FLEXIBLE_PRICE_EMAIL_BODY,
+    FlexiblePriceStatus,
+)
+from flexiblepricing.messages import FlexiblePriceStatusChangeMessage
+from ecommerce.discounts import DiscountType
+from ecommerce.models import Product
+
+log = logging.getLogger(__name__)
+
+RECIPIENT_VARIABLE_NAMES = {
+    "PreferredName": "preferred_name",
+    "Email": "email",
+}
+
+
+def generate_flexible_price_email(flexible_price):
+    """
+    Generates the email subject and body for a FinancialAid status update. Accepted statuses are
+    FlexiblePriceStatus.APPROVED and FlexiblePriceStatus.PENDING_MANUAL_APPROVAL (documents have been received).
+    Args:
+        flexible_price (FinancialAid): The FinancialAid object in question
+    Returns:
+        dict: {"subject": (str), "body": (str)}
+    """
+    if flexible_price.status == FlexiblePriceStatus.APPROVED:
+        courserun = CourseRun.objects.filter(
+            course=flexible_price.courseware_object
+        ).first()
+        product = Product.objects.get(object_id=courserun.id)
+        price = DiscountType.get_discounted_price(
+            [flexible_price.tier.discount], product
+        )
+        message = FLEXIBLE_PRICE_APPROVAL_MESSAGE.format(
+            program_name=flexible_price.courseware_object.title, price=price
```

Comment:
Should we rename this to something more generic to reflect both courses and programs?